src/devices/src/virtio/vsock/unix/muxer.rs

```diff
@@ -143,16 +143,22 @@ impl VsockChannel for VsockMuxer {
                         .set_flags(0)
                         .set_buf_alloc(0)
                         .set_fwd_cnt(0);
+                    self.rxq.pop().unwrap();
                     return Ok(());
                 }
 
                 // We'll defer building the packet to this connection, since it has something
                 // to say.
                 MuxerRx::ConnRx(key) => {
                     let mut conn_res = Err(VsockError::NoData);
+                    let mut do_pop = true;
```

Comment:
> it is necessary to keep the connection inside the muxer rxq after handling a pending RX instruction for that specific connection

The purpose of this change is to keep the connection inside the `rxq`, in the situation where there are multiple flags pending.

Previously to this, the muxer would permanently pop a connection from the `rxq` even if it had multiple flags pending, which made the unhandled flag to be dropped and lost forever. If that flag would have been a credit update, the flow control would be broken.

I agree that instructions for one connection would be prioritized, but since the instructions are flag based and multiple similar instructions cannot be queued at the same time, I don't think that there is a significant impact on neighboring connections. 
We could implement a _fair_ scheduler, but we also need to keep in mind that this code path is executed very often and we need an efficient way to look-up flags and pop elements.

If you have a proposal on this, let's discuss it.